File: 100-same-tree/same-tree.java
class Solution {

    void helper(TreeNode node , List<Integer> list) {
        if(node == null) {
            list.add(-1);
            return;
        }

        helper(node.left , list);
        list.add(node.val);
        helper(node.right , list);
        list.add(node.val);
    
    }

    public boolean isSameTree(TreeNode p, TreeNode q) {
        List<Integer> list1 = new ArrayList<>();
        if(p != null) {
            list1.add(p.val);
            helper(p , list1);
        }
        List<Integer> list2 = new ArrayList<>();
        if(q != null) {
            list2.add(q.val);
            helper(q , list2);
        }
        
        return list1.equals(list2);
    }
}

File: 100-same-tree/same-tree.cpp

class Solution {
    void sameTree(TreeNode* root, vector<string>& list) {
        queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()) {
            TreeNode* node = q.front();
            q.pop();

            if (node == nullptr) {
                list.push_back("null");
                continue;
            }

            list.push_back(to_string(node->val));
            q.push(node->left);
            q.push(node->right);
        }
    }

public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        vector<string> list1, list2;
        sameTree(p, list1);
        sameTree(q, list2);

        return list1 == list2;
    }
};
